moped-editor/src/views/projects/projectView/ProjectStatusBadge.js

```diff
@@ -0,0 +1,242 @@
+    },
+    "post-construction": {
+      color: white,
+      background: backgroundColors.success,
+      icon: PlayCircleOutlineOutlinedIcon,
+    },
+    potential: {
+      color: primary,
+      background: backgroundColors.warning,
+      icon: RemoveCircleOutlineOutlinedIcon,
+    },
+    canceled: {
+      color: white,
+      background: backgroundColors.error,
+      icon: CancelOutlinedIcon,
+    },
+    "on hold": {
+      color: primary,
+      background: backgroundColors.default,
+      icon: PauseCircleOutlineOutlinedIcon,
+    },
+    complete: {
+      color: white,
+      background: backgroundColors.info,
+      icon: CheckCircleOutlineOutlinedIcon,
+    },
+    default: {
+      color: primary,
+      background: defaultBackgroundColor,
+      icon: defaultIcon,
+    },
+  };
+
+  return styleMapping?.[phase] ?? styleMapping.default;
+};
+
+/**
+ * Picks the right color based on the status-phase combination
+ */
+const useFontColorStyles = makeStyles(theme => ({
+  root: {
+    // Find text color
+    color: props => getStyle(theme, (props?.phase ?? "").toLowerCase()).color,
+  },
+}));
+
+/**
+ * Picks the right chip color based on status-phase combination
+ */
+const useChipStyles = makeStyles(theme => ({
+  root: {
+    marginLeft: "1.5rem",
+    fontWeight: "500",
+    fontSize: "16px",
+    borderRadius: "2rem",
+    height: "2.5rem",
+    padding: ".5rem",
+
+    // Find background color
+    backgroundColor: props => getStyle(theme, props?.phase ?? "").background,
+  },
+}));
+
+/**
+ * Renders a chip
+ * @param {number} status - The status id of the current project
+ * @param {string} phase - The current phase name of the project
+ * @param {Object} projectStatuses - A dictionary of all available project statuses
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const ProjectStatusBadge = ({ status, phase, projectStatuses }) => {
+  /**
+   * Returns the label given a status-phase combination
+   * @param {number} status - The status id number
+   * @param {string} phase - The name of the phase
+   * @returns {string}
+   */
+  const getComponentMapName = (status, phase) =>
+    status > 1 // is not active?
+      ? // Then it can be found as a status
+        (
+          projectStatuses.find(s => s.status_id === status)?.status_name ??
+          String(phase)
+        ) // if not, default to phase
+          .toLowerCase()
+      : // if it's active then it's a phase for sure
+        String(phase).toLowerCase();
+
+  /**
+   * Bundle status properties object
+   */
+  const statusProperties = {
+    status: status,
+    phase: getComponentMapName(status, phase), // It's here so it can be corrected by status id
+  };
+
+  /**
+   * Generate font, chip and icon classes
+   */
+  const chipClasses = useChipStyles(statusProperties);
+  const iconClasses = useFontColorStyles(statusProperties);
+
+  // If we don't have a status, then do not render.
+  if (!!!status) return null;
```

Comment:
Why is it beneficial to use `!!!` over `!`?

Reply:
It's a little too fancy and maybe unnecessary, it's an inverted truthy operation... or an ugly catch-all method. We are not assuming status is a string, null, empty array, etc.

https://developer.mozilla.org/en-US/docs/Glossary/Truthy

Reply:
thanks, I was just under the impression that !!! was functionally similar to ! and was wondering if there was a different behavior from one to the other.

Reply:
Yeah this is really small stylistic thing, but there is no difference between `!!!` and `!` in the way truthiness gets evaluated. My opinion is that it might be confusing, but again, it doesn't really matter.
TIL: https://stackoverflow.com/questions/21154510/the-use-of-the-triple-exclamation-mark

Reply:
This is a zen moment! 🙇 I will have it changed in a new PR, I think there are other instances of this in the code. CDN avatar iirc. Thanks chia, mateo!